### PR TITLE
feature/VETHUB-50

### PR DESCRIPTION
**Pull request by Mohamad11Dab**
Introducing a `Timed Mode` alongside the existing `Tutor Mode.` In the Tutor Mode, users continue to receive feedback immediately after answering each question. The new Timed Mode, however, aggregates feedback and presents it only after the user has completed all questions within the quiz.
for testing : 
run the command:
`yarn run test`

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -8,11 +8,18 @@ import { useQuizStore } from "./store/QuizStore";
 const quizQuestions = ref(0);
 const questionsQueue = useQuizStore();
 const quizStarted = ref<boolean>(false);
+const quizMode = ref<string>("Tutor");
 
-const handleStartQuiz = (questionAmount: number) => {
+const handleStartQuiz = ({
+  questionAmount,
+  mode,
+}: {
```
Comment (anzau) [inline src/App.vue]: I would add the types in the `.d.ts` file

File: src/store/QuizStore.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { MCQuestion, QuestionState } from "@/types/MCQ";
 import { defineStore } from "pinia";
 
 type Stat = "correct" | "skipped" | "attempts" | "selectedValue";
+type QuizMode = string;
```
Comment (anzau) [inline src/store/QuizStore.ts]: Not sure if this is needed?

Comment (freshc0w) [reply src/store/QuizStore.ts]: Converted it to a type in .d.ts file.

File: src/App.vue (hunk 1 above)
Comment (freshc0w) [reply src/App.vue]: Done

**Review by freshc0w — APPROVED** [review]
Because this PR has been up for a while now, and a lot of conflicts have been accumulated compared with develop, I've fixed up Anzah comments and fixed merge conflicts so this may be merged ASAP.

Overall functionality LGTM

**Pull request merged**